commit 6c95c2eebead155776b68013283c6bb712fb3755
Author: Nicholas Mc Guire <hofrat@osadl.org>
Date:   Wed Nov 21 19:06:12 2018 +0100

    gpio: mt7621: report failure of devm_kasprintf()
    
    [ Upstream commit 59d646c775d6ae688ee90fda9f2a4270c47b7490 ]
    
    kasprintf() may return NULL on failure of internal allocation thus the
    assigned  label  is not safe if not explicitly checked. On error
    mediatek_gpio_bank_probe() returns negative values so -ENOMEM in the
    (unlikely) failure case should be fine here.
    
    Fixes: 4ba9c3afda41 ("gpio: mt7621: Add a driver for MT7621")
    Signed-off-by: Nicholas Mc Guire <hofrat@osadl.org>
    Reviewed-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
    Acked-by: Sean Wang <sean.wang@kernel.org>
    Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/gpio/gpio-mt7621.c b/drivers/gpio/gpio-mt7621.c
index d72af6f6cdbd..1ec95bc18f5b 100644
--- a/drivers/gpio/gpio-mt7621.c
+++ b/drivers/gpio/gpio-mt7621.c
@@ -244,6 +244,8 @@ mediatek_gpio_bank_probe(struct device *dev,
 	rg->chip.of_xlate = mediatek_gpio_xlate;
 	rg->chip.label = devm_kasprintf(dev, GFP_KERNEL, "%s-bank%d",
 					dev_name(dev), bank);
+	if (!rg->chip.label)
+		return -ENOMEM;
 
 	ret = devm_gpiochip_add_data(dev, &rg->chip, mtk);
 	if (ret < 0) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1104_6c95c2eebead155776b68013283c6bb712fb3755.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1104_6c95c2eebead155776b68013283c6bb712fb3755.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
